### PR TITLE
xtest: add SM2 perf test

### PR DESCRIPTION
**Pull request by yuzexiyzx**
add perf test for SM2 algorithm

### CODE REVIEW

File: host/xtest/asym_perf.c [hunk 1]
```diff
@@ -904,6 +904,25 @@ int asym_perf_runner_cmd_parser(int argc, char *argv[])
 				main_algo = ALGO_X25519;
 				width_bits = 256;
 				mode = MODE_GENKEYPAIR;
+			} else if (!strcasecmp(argv[i], "SM2_GENKEYPAIR")) {
```
Comment (etienne-lms) [inline host/xtest/asym_perf.c:907]: Don't you need to provide the `main_algo` info to generate an SM2 keypair?

File: host/xtest/asym_perf.c [hunk 2]
```diff
@@ -957,7 +976,7 @@ int asym_perf_runner_cmd_parser(int argc, char *argv[])
 		}
 	}
 
-	if (mode == MODE_GENKEYPAIR)
+	if (mode == MODE_GENKEYPAIR || main_algo >= ALGO_SM2_PKE)
```
Comment (etienne-lms) [inline host/xtest/asym_perf.c:979]: prefer explicit matching `main_algo` values:
```c
	if (mode == MODE_GENKEYPAIR || main_algo >= ALGO_SM2_PKE ||
	    main_algo >= ALGO_SM2_DSA_SM3)

**Review by etienne-lms — COMMENTED** [review]
@yuzexiyzx, `usage()` needs to be updated for these new cmdline options.

File: host/xtest/asym_perf.c (hunk 1 above)
Comment (yuzexiyzx) [reply host/xtest/asym_perf.c:909]: sloved

Comment (yuzexiyzx) [reply host/xtest/asym_perf.c:909]: solved

File: host/xtest/asym_perf.c (hunk 2 above)
Comment (yuzexiyzx) [reply host/xtest/asym_perf.c]: solved

**Comment by yuzexiyzx** [comment]
@etienne-lms All comments have been comments.

**Comment by yuzexiyzx** [comment]
@etienne-lms All comments have been solved.

**Review by etienne-lms — COMMENTED** [review]
LGTM.
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>` if this new comment is out of scope.

This change modifies asym performance test TA API, but I think it's ok we don't enfore backward compat on OP-TEE Test TAs ABI.

**Comment by yuzexiyzx** [comment]
@etienne-lms This is used for both TEE_TYPE_SM2_DSA_KEYPAIR and TEE_TYPE_SM2_PKE_KEYPAIR
image

**Comment by yuzexiyzx** [comment]
@jforissier All comments have been solved.

**Comment by etienne-lms** [comment]
@jforissier, LGTM. Note the test TA API change. I think it's ok we don't enfore backward compat on OP-TEE Test TAs ABI. Ok with you?

**Comment by jforissier** [comment]
> @jforissier, LGTM. Note the test TA API change. I think it's ok we don't enfore backward compat on OP-TEE Test TAs ABI. Ok with you?

Yes it's fine.

**Pull request merged**